src/Yesod/Auth/OAuth2/Dispatch.hs

```diff
@@ -109,7 +110,7 @@ withCallbackAndState name oauth2 csrf = do
   callback <- maybe (throwError $ InvalidCallbackUri uri) pure $ fromText uri
```

Comment:
Should be move the defaulting up here such that `oauth2RedirectUri` being `Just` will avoid the possibility of `throwError` altogether?

Reply:
For example,

```hs
withCallbackAndState name oauth2 csrf = do
  callback <- maybe getDefaultCallback pure oauth2RedirectUri oauth2

  pure
    oauth2
      { oauth2RedirectUri = Just callback

getDefaultCallback = do
  uri <- ($ PluginR name ["callback"]) <$> getParentUrlRender
  maybe (throwError $ InvalidCallbackUri uri) pure $ fromText uri
```